mk/sketch_sources.mk

```diff
@@ -21,6 +21,48 @@ SKETCHOBJS := $(addsuffix .o,$(basename $(SKETCHOBJS)))
 include $(MAKE_INC)
 LIBTOKENS := $(LIBRARIES)
 
+
+# HAL and board specific libraries are included here.
+LIBTOKENS += \
+	AP_HAL \
+	AP_HAL_Empty
+
+ifeq ($(HAL_BOARD),HAL_BOARD_APM1)
+LIBTOKENS += \
+	AP_HAL_APM
+endif
+
+ifeq ($(HAL_BOARD),HAL_BOARD_APM2)
+LIBTOKENS += \
+	AP_HAL_APM
+endif
```

Comment:
I don't expect this kind of change to be backported so maybe we should just not add APM1 and APM2 here?